pkg/scanners/helm/test/scanner_test.go

```diff
@@ -0,0 +1,138 @@
+package test
+
+import (
+	"context"
+	"io"
+	"os"
+	"path/filepath"
+	"sort"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/aquasecurity/defsec/pkg/scanners/helm"
+)
+
+func Test_helm_scanner_with_archive(t *testing.T) {
+
+	tests := []struct {
+		testName    string
+		chartName   string
+		path        string
+		archiveName string
+	}{
+		{
+			testName:    "Parsing tarball 'mysql-8.8.26.tar'",
+			chartName:   "mysql",
+			path:        filepath.Join("testdata", "mysql-8.8.26.tar"),
+			archiveName: "mysql-8.8.26.tar",
+		},
+	}
+
+	for _, test := range tests {
+		t.Logf("Running test: %s", test.testName)
+
+		helmScanner := helm.New(test.chartName)
+
+		testTemp := t.TempDir()
+		testFileName := filepath.Join(testTemp, test.archiveName)
+		require.NoError(t, copyArchive(test.path, testFileName))
+
+		testFs := os.DirFS(testTemp)
+		results, err := helmScanner.ScanFS(context.TODO(), testFs, ".")
+		require.NoError(t, err)
+		require.NotNil(t, results)
+
+		failed := results.GetFailed()
+		assert.Equal(t, 32, len(failed))
+
+		visited := make(map[string]bool)
+		var errorCodes []string
+		for _, result := range failed {
+			id := result.Flatten().RuleID
+			if _, exists := visited[id]; !exists {
+				visited[id] = true
+				errorCodes = append(errorCodes, id)
+			}
+		}
+		assert.Len(t, errorCodes, 19)
```

Comment:
Would be cool to have a test that checks for start/end lines as it seems to be broken everywhere in the other scanners :(

Reply:
yeah, could add that when I start getting line numbers working